hooks/generatePdf.js

```diff
@@ -0,0 +1,20 @@
+const puppeteer = require('puppeteer');
+const path = require('path');
+
+module.exports = {
+    'generate:after': generatePdf
+};
+
+async function generatePdf(generator) {
+    const targetDir = generator.targetDir;
+    const parameters = generator.templateParams;
+    if (!parameters || parameters.pdf === undefined) return;
```

Comment:
you removed by accident `const browser = await puppeteer.launch();`


I also think it would be better to replace `parameters.pdf === undefined` with `parameters.pdf !== 'true'` and remove line 16. The generator always gets string, so with the current implementation, so if someone would pass `-p pdf=false` then puppeteer would unnecessarily start a browser.

You can test your work easily on local like this:
- just install generator `npm i -g @asyncapi/generator`
- then run in from the root of your fork like `ag https://bit.ly/asyncapi ./ -o test/output -p pdf=true` or even `ag test/spec/asyncapi.yml ./ -o test/output -p pdf=true` (so you can play with test file from test/spec/asyncapi.yml in case you want to see how the behaviour of template changes)

Reply:
@derberg Thanks for the instructions, In hindsight, I should have run it locally before even making a PR, I do apologise. I've run it locally with the changes you've asked. Although with the `pdf=true` in the command u mentioned, I don't see the difference in the `test/output` file. Is there something wrong or am I looking at the wrong place?

Reply:
no worries. You do not get pdf because in your code you expect boolean `true`, while as I wrote, CLI passes always string, so `true` is actually a string `'true'`. I made a code suggestion.

I'm more affraid with the quality of the generated PDF. It has all the data collapsed by default, so if you get a payload with object, you cannot see data of the object. Would make sense to use puppeteer here and before pdf generation, perform a click on collapse/expand chevron across the entire document to get all those expanded. You know what I mean?

Reply:
So what you're trying to say is, to make a preview modal in the html, to preview the pdf before it's generated?

Reply:
no, let me explain more in details:

this is how the message under channel is shown, in HTML and then in PDF
<img width="784" alt="Screenshot 2020-10-07 at 09 15 15" src="https://user-images.githubusercontent.com/6995927/95299456-0bb14300-087e-11eb-960b-d6beb13f8037.png">

so it is good, very good, as payload and headers are expanded. Problem is that in HTML they are not expanded in both, example and messages section:
<img width="711" alt="Screenshot 2020-10-07 at 09 15 06" src="https://user-images.githubusercontent.com/6995927/95299525-28e61180-087e-11eb-98bd-2c2011bd07da.png">
<img width="358" alt="Screenshot 2020-10-07 at 09 14 55" src="https://user-images.githubusercontent.com/6995927/95299563-37ccc400-087e-11eb-9708-633ab6da27bb.png">

For HTML it is not a problem as user can click and expand, but yeah, not in case of PDF. 
I suggest we use the power of puppeteer here. It has a rich API to interact with a browser, many people use it to write UI tests (I did in the past too). 
So before you invoke `await page.pdf({ format: 'A4', path: `${targetDir}/index.pdf` });` you can interact with the page and perform `clicks` on those chevrons to expand those `payload` and `headers` sections.

something like this should work to expand all payload examples 
```
await page.$$eval('div.payload-examples > div.js-prop', links => links.forEach(link => link.click()))
```
For headers you would have to eval on `div.header-examples`, I think 😄  and not sure about `Messages` section, I think that maybe [here](https://github.com/asyncapi/html-template/blob/master/partials/messages.html#L8) we need to add some custom class name that could be use in selectors.

with UI operations like this remember that we need to make sure that the page is fully loaded before we execute clicks so modification like this would be needed 
```
await page.goto(`file://${path.join(targetDir, 'index.html')}`, { waitUntil: 'networkidle0' });
```

also add an option to generate background in pdf, should look like 
```
 await page.pdf({ format: 'A4', path: `${targetDir}/index.pdf`, printBackground: true });
```

Reply:
Ah I see what you mean now. 
I did confirm that `hearders-examples` expands the `Headers`, but for the `Message`, adding a `message-example` and making a similar `page.$$eval` doesn't expand the `payload` and `header` chevrons. I've tried placing the class elsewhere but it still didn't work.